src/js/wall/__tests__/LoginForm.test.tsx

```diff
@@ -1,74 +1,78 @@
-
         window.virtool = {
             b2c: {
                 use: false,
             },
         };
     });
 
+    afterEach(() => nock.cleanAll());
+
     it("should call mutate() with correct values when submitted", async () => {
-        renderWithProviders(<LoginForm error="" />);
+        const username = "test_Username";
+        const password = "Password";
+
+        const scope = nock("http://localhost")
+            .post("/api/login", { username: username, password: password, remember: true })
+            .reply(200);
+
+        renderWithProviders(<LoginForm />);
 
         const usernameField = screen.getByLabelText("Username");
-        await userEvent.type(usernameField, "test_Username");
-        expect(usernameField).toHaveValue("test_Username");
+        await userEvent.type(usernameField, username);
+        expect(usernameField).toHaveValue(username);
 
         const passwordField = screen.getByLabelText("Password");
-        await userEvent.type(passwordField, "Password");
-        expect(passwordField).toHaveValue("Password");
+        await userEvent.type(passwordField, password);
+        expect(passwordField).toHaveValue(password);
 
         expect(screen.getByLabelText("Remember Me")).toHaveAttribute("data-state", "unchecked");
         await userEvent.click(screen.getByLabelText("Remember Me"));
         expect(screen.getByLabelText("Remember Me")).toHaveAttribute("data-state", "checked");
 
         await userEvent.click(screen.getByRole("button", { name: "Login" }));
-        await waitFor(() =>
-            expect(mockLoginMutation.mutate).toHaveBeenCalledWith(
-                { username: "test_Username", password: "Password", remember: true },
-                expect.any(Object),
-            ),
-        );
+
+        scope.isDone();
     });
 
     it("should display error message on login failure", async () => {
-        mockLoginMutation.isError = true;
+        const username = "test_Username";
+        const password = "Password";
+        const errorMessage = "An error occurred during login";
 
-        renderWithProviders(<LoginForm error="Invalid username or password" />);
+        const scope = nock("http://localhost")
+            .post("/api/login", { username: username, password: password, remember: false })
+            .reply(400);
 
-        await userEvent.type(screen.getByLabelText("Username"), "test_Username");
-        await userEvent.type(screen.getByLabelText("Password"), "Password");
+        renderWithProviders(<LoginForm />);
+
+        await userEvent.type(screen.getByLabelText("Username"), username);
+        await userEvent.type(screen.getByLabelText("Password"), password);
         await userEvent.click(screen.getByRole("button", { name: "Login" }));
 
-        await waitFor(() => expect(screen.getByText("Invalid username or password")).toBeInTheDocument());
+        await waitFor(() => {
+            expect(screen.getByText(errorMessage)).toBeInTheDocument();
```

Comment:
the `FindBy` family of async DOM queries combines the get by and wait for in a convenient package. https://testing-library.com/docs/dom-testing-library/api-async/